esmvaltool/cmorizers/data/formatters/datasets/esacci_soilmoisture.py

```diff
@@ -0,0 +1,163 @@
+"""ESMValTool CMORizer for ESACCI-SOILMOISTURE data.
+
+Tier
+    Tier 2: other freely-available dataset.
+
+Source
+    ftp://anon-ftp.ceda.ac.uk/neodc/esacci/soil_moisture/data/
+
+Last access
+    20240626
+
+Download and processing instructions
+    Download the data from:
+      daily_files/COMBINED/v08.1/
+      ancillary/v08.1/
+    Put all files under a single directory (no subdirectories with years).
+    in ${RAWOBS}/Tier2/ESACCI-SOILMOISTURE
+
+"""
+
+import glob
+import logging
+import os
+from datetime import datetime
+import iris
+from esmvalcore.preprocessor import concatenate, monthly_statistics
+from cf_units import Unit
+
+from ...utilities import (
+    fix_var_metadata,
+    fix_dim_coordnames,
+    fix_bounds,
+    save_variable,
+    set_global_atts,
+    roll_cube_data
+)
+
+logger = logging.getLogger(__name__)
+
+
+def fix_coords(cube):
+    """Fix coordinates to CMOR standards.
+
+    Fixes coordinates eg time to have correct units, bounds etc;
+    longitude to be CMOR-compliant 0-360deg; fixes some attributes
+    and bounds - the user can avert bounds fixing by using supplied
+    arguments; if bounds are None they will be fixed regardless.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        data cube with coordinates to be fixed.
+
+
+    Returns
+    -------
+    cube: iris.cube.Cube
+        data cube with fixed coordinates.
+    """
+    # First fix any completely missing coord var names
+    fix_dim_coordnames(cube)
+    # Fix individual coords
+    for cube_coord in cube.coords():
+        # Fix time
+        if cube_coord.var_name == 'time':
+            logger.info("Fixing time...")
+            cube.coord('time').convert_units(
+                Unit('days since 1970-01-01T00:00:00+00:00',
+                     calendar='proleptic_gregorian'))
+
+        # Fix latitude
+        if cube_coord.var_name == 'lat':
+            logger.info("Fixing latitude...")
+            cube = iris.util.reverse(cube, cube_coord)
+
+        # Fix longitude
+        if cube_coord.var_name == 'lon':
+            logger.info("Fixing longitude...")
+            if cube_coord.ndim == 1:
+                if cube_coord.points[0] < 0. and \
+                        cube_coord.points[-1] < 181.:
+                    cube_coord.points = \
+                        cube_coord.points + 180.
+                    cube.attributes['geospatial_lon_min'] = 0.
+                    cube.attributes['geospatial_lon_max'] = 360.
+                    nlon = len(cube_coord.points)
+                    roll_cube_data(cube, nlon // 2, -1)
```

Comment:
```suggestion
```

This causes the data to be longitude -180...180, but with Europe at -180 (bad!). See other comment about `cube.intersection` for a fix.

Reply:
Please remove this part, it is not necessary anymore (you can simply click on "Commit suggestion" above) 👍